internal/transport/transport_test.go

```diff
@@ -2045,6 +2047,25 @@ func (s) TestClientDecodeHeaderStatusErr(t *testing.T) {
 				"peer header list size exceeded limit",
 			),
 		},
+		{
+			name: "bad status in grpc mode",
+			metaHeaderFrame: &http2.MetaHeadersFrame{
+				Fields: []hpack.HeaderField{
+					{Name: "content-type", Value: "application/grpc"},
+					{Name: "grpc-status", Value: "0"},
+					{Name: ":status", Value: "504"},
+				},
+				HeadersFrame: &http2.HeadersFrame{
+					FrameHeader: http2.FrameHeader{
+						StreamID: 0,
+					},
+				},
+			},
+			wantStatus: status.New(
+				codes.Unavailable,
+				"Gateway Timeout: HTTP status code 504; transport: missing content-type field",
```

Comment:
`transport: missing content-type field` looks incorrect, since a valid `content-type` header is present.

Also maybe the main part of the error message can be clearer?  E.g.

`unexpected HTTP status code received from server: 504 (Gateway Timeout)`

Reply:
That does seem wrong. I've updated the messaging!